app/models/user.rb

```diff
@@ -0,0 +1,17 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: :author_id
+  has_many :comments, foreign_key: :author_id
+  has_many :likes, foreign_key: :author_id
+
+  def recent_posts
+    posts.order(created_at: :desc).limit(3)
+  end
+
+  def update_posts_counter
+    update(posts_counter: posts.count)
+  end
+
+  def update_comments_counter
+    update(comments_counter: comments.count)
+  end
```

Comment:
- [ ] [OPTIONAL] in as much as the comment belongs to a user it also belong to a post, so this method will be best left in the post model, I am making it optional because you already have it in the post model, so remove it here